tornado/ioloop.py

```diff
@@ -932,6 +963,7 @@ def set_blocking_signal_threshold(self, seconds, action):
             signal.signal(signal.SIGALRM,
                           action if action is not None else signal.SIG_DFL)
 
+    @cython.locals(timeout = _Timeout, now = cython.double)
```

Comment:
ncallbacks is unnecessary (cython infers its type), and poll_timeout is passed to python functions so it needs to be an object to avoid having to create one